### PR TITLE
30-Update-Base-Classes-with-new-kinematic-model

### PR DESCRIPTION
**Pull request by e10harvey**
Supersedes https://github.com/sandialabs/OpenCSP/pull/105

### CODE REVIEW

**Comment by braden6521** [comment]
@e10harvey, how should I handle reviewing this?

**Comment by e10harvey** [comment]
@braden6521, @mhhwang1: This is ready for reviews.

File: contrib/scripts/sensitive_strings.py
Comment (braden6521) [inline contrib/scripts/sensitive_strings.py:1]: Does anyone know why the sensitive_strings file is changed as part of this?

**Review by braden6521 — APPROVED** [review]
@e10harvey, like we talked about, this is too big to review thoroughly. I looked through the non-unit test changes as best I can right now. It will likely need some clean-up later, but it looks good enough now to merge in and get this off our plates.

**Pull request merged**